### ISSUE TITLE
Graph ML 

### ISSUE THREAD

**Issue by Anindyadeep** [microsoft/ML-For-Beginners#500]
Hello all!
I want to contribute to Graph ML in PyTorch/PyG as a part of the course. I have been studying Graph Machine Learning for some months, and due to some lack of resource, I wanna contribute here. 

1. I will start from very scratch, from simple intuitive examples I will make people understand the difference between the traditional ML/DL paradigm to Graph based paradigm. Because, I have seen, people facing the problem in this transition. 
2. Continuing to that I will move towards simple model making, and the basic maths required to understand the inside working of the model. 
3. Finally solving problems using real life dataset. 

If this looks good, please let me know. I will be happy to contribute here. Thank you.

**Comment by jlooper** [comment]
hi, thank you for reaching out. This curriculum is in a pretty fixed state such that we won't add substantively to it; in addition, we avoided using PyTorch/TensorFlow in this ML curriculum to narrow its focus to classic ML. However, there's a new AI curriculum being created by @shwars and this topic might fit in better there.

**Comment by Anindyadeep** [comment]
Thanks, I will surely reach out there. Also, I wanna add something more. As you mentioned that this repo is focusing more on classical ML algorithm, and I was also doing a project lately, of making different ML algorithm from scratch. So, can I write here also, like how to make traditional ML models from numpy by understanding the inner workings of those? Please let me know. Also thank you for reaching out.

**Comment by shwars** [comment]
@Anindyadeep thank you for reaching out! can you share with me the kind of examples you have in mind? This curriculum (as well as AI curriculum) are intended as an introductory course, so we need to balance the complexity of material. We also wanted to have limited amount of content so that people can learn within one semester timeframe.
The idea of including algorithm implementation from scratch into this curriculum sounds attractive, but @jlooper is the one to decide here. Maybe it would make sense to add extra level of "optional" complexity for those who want to dig deeper, especially if you can stick to the same datasets/samples that are implemented in this curriculum via sklearn.

**Comment by Anindyadeep** [comment]
@shwars sir. I completely understand about the situation of the complexity and the time frame constraints. So here is the plan, I will start from the very very basics, like the tools of understanding the algorithms, which are nothing but some high school math revision. Theoretical ML could be really hard, given there are so many concepts, but at the same time some of those concepts could be translated to this high school level of simplicity. As those includes some revision of matrix, differentiation, some basic probability and some miscellaneous concepts.

Once covered we can provide the readers with articles classified under some levels of complexity.

Level 0 can be the revisions of some mathematical tools for ML. Also including some basic numpy operations.

Level 1 (Supervised ) could be the simple linear regression and logistic regression (binaray classification)

Level 2 could include multi class logistic regression, as softmax is kinda overwhelming. Also naive bayes. So till here we completed some supervised ml algorithms. 

Level 3 (last) Here we can cover 1 or 2 unsupervised ML algorithmis. Like PCA or any other. 

These levels are kinda choice based  so that readers could follow according to their interests and level of complexity.

I will try my very best, not to overwhelm the things, and keep them simple, such that the  comfort of the reader could be maintained. 

So this is all about my plan on the curriculum. If this convinces you, then let me know, I will show some demos also. Thank you.

**Comment by jlooper** [comment]
hi, I don't think we want to add anything major to this ML curriculum, as its design is fixed and complete. The outline here however seems to coincide in part with content that is already covered in the current ML curriculum (this repo). 

In general, the design of this curriculum has avoided deep theoretical dives in favor of project-based learning using Scikit-Learn. 

However, I wouldn't be averse to adding one 'sidecar lesson' on some of the mathematical underpinnings that will be helpful to understand when learning ML. I would want you to provide me a very clear and detailed outline of the topics you want to cover in that lesson along with a writing sample (it can be an introductory paragraph). Each item will need an infographic illustration to align to our style, so I will have to provide that myself. For this reason I need to be clear on managing the scope of the lesson.

If you are interested in that, we might proceed. thanks!

**Comment by Anindyadeep** [comment]
Thank you for reaching out. Yeah definitely, I would love to contribute. So you want me to provide, the deep details of the topic, as well as some of kind of sample of a lesson, with info-graphics right. If so, then, I will try my best to provide you by tomorow.
